source/mbed_platform.cpp

```diff
@@ -32,6 +36,30 @@ static bool timeIsInPeriod(minar::platform::tick_t start, minar::platform::tick_
 namespace minar {
 namespace platform {
 
+namespace test {
+#if YOTTA_CFG_MINAR_TEST_CLOCK_OVERFLOW
+    #define BUFFER_SIZE 128
+    static uint32_t sleep_until_buf[BUFFER_SIZE];
+    static uint32_t sleep_until_buf_tail = 0;
+
+    uint32_t *get_sleep_until_buf(void) {
+        return (uint32_t *) &sleep_until_buf;
+    }
+
+    uint32_t get_sleep_until_buf_tail(void) {
+        return sleep_until_buf_tail;
+    }
+
+    uint32_t inc_sleep_until_buf_tail(void) {
+        if (sleep_until_buf_tail < BUFFER_SIZE-1) {
+            return ++sleep_until_buf_tail;
+        } else {
+            return -1;
```

Comment:
Returning -1 for an `uint32_t` is a bit weird :) Returning `BUFFER_SIZE` or a predefined `BUFFER_FULL` constant is a better idea. Do you intend to check the return code of this function anywhere ?